extensions/devui/runtime/src/main/java/io/quarkus/devui/runtime/comms/JsonRpcRouter.java

```diff
@@ -202,15 +202,17 @@ private void routeToRuntimeMethod(JsonRpcRequest jsonRpcRequest, JsonRpcResponse
         }
         uni.subscribe()
                 .with(item -> {
-                    if (item != null && JsonRpcMessage.class.isAssignableFrom(item.getClass())) {
-                        JsonRpcMessage<?> jsonRpcMessage = (JsonRpcMessage<?>) item;
-                        Object response = jsonRpcMessage.getResponse();
-                        if (jsonRpcMessage.isAlreadySerialized()) {
+                    if (item != null && Map.class.isAssignableFrom(item.getClass())) {
+                        Map map = (Map) item;
+                        if (map.size() == 3 && map.containsKey("alreadySerialized") && map.containsKey("messageType")
+                                && map.containsKey("response") && map.get("alreadySerialized").equals("true")) {
```

Comment:
I had added the `alreadySerialized` flag to `JsonRpcMessage` to allow for custom serialization, if we want `Map` to represent a custom response format it might be enough to check for `response` and `messageType` perhaps.

Also we can consider removing the flag from `JsonRpcMessage` altogether, as it won't be used anywhere else after this, to keep it clean.

Reply:
Sure, I mean having exactly `response` and `messageType` would already be a good-enough discriminator IMO, but if you want to make it even more explicit that's fine. At that point I would say let's put `alreadySerialized` as a boolean value instead of `String.valueOf(true)`.